### DISCUSSION TITLE
AI Assistant for self-hosted Supabase?

### DISCUSSION THREAD

**Discussion by brandonzylstra** [supabase/supabase#31149] [Questions] [answered]
I can't find any documentation about whether or not the assistant is available in the open source version of Supabase, or whether it is only available in the cloud service.  (I have not installed Supabase yet, so perhaps this is obvious to anyone who is already using it, but since _some_ —perhaps _many_— development tools that describe themselves as "open source" hold back features if you aren't using the cloud service, the answer is not obvious to me.)

Also, I found [other people who had similar questions](https://github.com/orgs/supabase/discussions/21967#discussioncomment-11453064) which have been unanswered for quite some time.  I think opening a new discussion is better than just adding an 👀 emoji to the previous discussion (in which this question was extremely tangential), as it's more likely that someone who can answer the question will notice it and reply.

**Comment by GaryAustin1** [comment, marked as answer]
https://supabase.com/features/ai-assistant
![image](https://github.com/user-attachments/assets/85f324bf-9637-442a-b77b-4dc892910aa2)

It seems their intent is that it is/will be with your own open AI key.
I don't find a quick link to how to do that though.

**Reply by brandonzylstra** [reply]
@GaryAustin1 thanks!

**Comment by volfadar** [comment]
For those of you searching for the environment variable to include in the docker-compose.yml file, you should add `OPENAI_API_KEY` under the `supabase-studio` service section
https://github.com/supabase/supabase/blob/f6d96174135fddc2c2bf8a7dd59225a02aee5559/docker/docker-compose.yml#L34

**Comment by pMertDogan** [comment]
Can we use any OpenAI-compatible provider other than OpenAI?
